sdk/cosmos/microsoft-azure-cosmos/src/main/java/com/azure/data/cosmos/internal/changefeed/implementation/PartitionSupervisorImpl.java

```diff
@@ -51,38 +51,40 @@ public PartitionSupervisorImpl(Lease lease, ChangeFeedObserver observer, Partiti
 
     @Override
     public Mono<Void> run(CancellationToken shutdownToken) {
-        PartitionSupervisorImpl self = this;
         this.resultException = null;
 
-        ChangeFeedObserverContext context = new ChangeFeedObserverContextImpl(self.lease.getLeaseToken());
+        ChangeFeedObserverContext context = new ChangeFeedObserverContextImpl(this.lease.getLeaseToken());
 
-        self.observer.open(context);
+        this.observer.open(context);
 
         ChangeFeedObserverCloseReason closeReason = ChangeFeedObserverCloseReason.UNKNOWN;
 
         try {
-            self.processorCancellation = new CancellationTokenSource();
+            PartitionSupervisorImpl self = this;
+            this.processorCancellation = new CancellationTokenSource();
 
             Thread processorThread = new Thread(new Runnable() {
```

Comment:
when you create threads someone has to manage the thread.

unmanaged threads in general are bad. as you won't know how many threads are created who is going to shut them down. when the java app is going to finish non-deamon threads will prevent a gracefull exit, etc.

When you use a Reactor Scheduler or even a pure ExecutorService  they will take care of the lifetime of your threads. When you close an executor it will shutdown all its underlying threads and resources.

we should rely on the reactor Scheduler for scheduling things but if you are using ExecutorService at least you should submit an instance of Runnable or Callable not a thread to the executor service.

Reply:
Fixed.
I'm keeping the executor scheduler for now (customer requested feature from the past) with the open mind to replace it with Reactor Scheduler.